site-packages/integralstor_common/rsync.py

```diff
@@ -0,0 +1,130 @@
+"""
+This file generates the rsync config.
+Assumed is that, xinetd is already installed and rsync deamon is enabled in xinetd
+
+Path of config : /etc/rsyncd.conf
+
+Default Config :
+[rsync_share_name]
+
+path=/pool/dataset/dir
+comment="Description of this share"
+list=true
+readonly = False, Selectable
+uid = integralstor (Selectable)
+gid = integralsto (Selectable)r
+
+"""
+
+import common,db
+
+def _generate_rsync_config():
+  try:
+    shares, err = load_shares_list()
+    if err:
+      raise Exception(err)
+    with open("/etc/rsyncd.conf","w") as f:
+      f.write("uid = integralstor\n")
+      f.write("gid = integralstor\n")
+      f.write("\n")
+      for share in shares:
+        f.write("[%s]\n"%share["name"])
+        f.write("\n")
+        f.write("path = %s\n"%share["path"])
+        f.write("comment = %s\n"%share["comment"])
+        list = 'true' if share["list"] else 'false'
+        readonly = 'true' if share["readonly"] else 'false'
+        f.write("list = %s\n"%list)
+        f.write("readonly = %s\n"%readonly)
+        f.write("\n")
+  except Exception, e:
+    return False, 'Error generating rsync config : %s'%str(e)
+  else:
+    return True, None
+
+def create_rsync_share(name,path,comment,list,readonly,uid,gid):
+  try:
+    db_path, err = common.get_db_path()
+    if err:
+      raise Exception(err)
+
+    check,err = db.read_single_row(db_path, "select * from rsync where name='%s'"%name)
+    if  check:
+      raise Exception("Share already exists.Use a different share name") 
+
+    cmd_list = []
+    cmd = ["insert into rsync (name,path,comment,list,readonly,uid,gid) values(?,?,?,?,?,?,?)", (name,path,comment,list,readonly,uid,gid)]
+    cmd_list.append(cmd)
+    ret, err = db.execute_iud(db_path, cmd_list)
+    if err:
+      raise Exception(err)
+    config ,err = _generate_rsync_config()
+    if err:
+      raise Exception(err)
+  except Exception, e:
+    return False, 'Error saving rsync config settings : %s'%str(e)
+  else:
+    return True, None
+
+def get_rsync_share_details(name):
+  
+  try:
+    db_path, err = common.get_db_path()
+    if err:
+      raise Exception(err)
+    cmd_list = []
+    check,err = db.read_single_row(db_path, "select * from rsync where name='%s'"%name)
+    if not check:
+      raise Exception("Share doesnot exists") 
+    return check,None
```

Comment:
Are we talking about line no : 79 ? If yes, then check is a local variable, else, that would involve declaring another global variable outside